kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/wire/store/provider/WireRecordStoreProvider.java

```diff
@@ -27,10 +28,30 @@ public interface WireRecordStoreProvider {
     /**
      * Opens or creates a {@link WireRecordStore} instance with the given name.
      * 
-     * @param name the store name
+     * @param name
+     *            the store name
      * @return the result {@link WireRecordStore}.
      * @throws KuraStoreException
      */
     public WireRecordStore openWireRecordStore(String name) throws KuraStoreException;
 
+    /**
+     * Adds a {@link ConnectionListener}
+     *
+     * @param listener
+     *            to add
+     *
+     * @since 2.5.0
+     */
+    public void addListener(ConnectionListener listener);
```

Comment:
We should document that the listener should close any currently open WireRecordStore instance, the same note as the MessageStoreProvider.